### PR TITLE
Vault integration: add repository create credential store method

### PR DESCRIPTION
**Pull request by mgaffney**

### CODE REVIEW

File: internal/credential/vault/repository_test.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
+package vault
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
```
Comment (louisruch) [inline internal/credential/vault/repository_test.go:8]: white space

File: internal/credential/vault/repository_credential_store.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	tokenExpires, err := renewedToken.TokenTTL()
+	if err != nil {
+		return nil, errors.Wrap(err, op, errors.WithMsg("unable to get vault token expiration"))
+	}
+
+	token, err := newToken(id, cs.token, tokenExpires)
+	if err != nil {
+		return nil, err
+	}
+	token.StoreId = id
```
Comment (louisruch) [inline internal/credential/vault/repository_credential_store.go:94]: `newToken` already set the `StoreId`

**Review by louisruch — APPROVED** [review]
Two minor comments otherwise this looks great!

File: internal/credential/vault/repository_test.go (hunk 1 above)
Comment (mgaffney) [reply internal/credential/vault/repository_test.go:8]: ty!

**Pull request merged**